jwql/instrument_monitors/common_monitors/dark_monitor.py

```diff
@@ -0,0 +1,988 @@
+#! /usr/bin/env python
+
+"""This module contains code for the dark current monitor, which
+performs some basic analysis to check whether the dark current behavior
+for the most recent set of input files is consistent with that from
+past files.
+
+If enough new files for a given instrument/aperture combination
+(currently the files must be identified as dark current files in the
+``exp_type`` header keyword) are present in the filesystem at the time
+the ``dark_monitor`` is called, the files are first run through the the
+appropriate pipeline steps to produce slope images.
+
+A mean slope image as well as a standard deviation slope image is
+created by sigma-clipping on a pixel by pixel basis. The mean and
+standard deviation images are saved to a fits file, the name of which
+is entered into the ``<Instrument>DarkCurrent`` database table.
+
+The mean slope image is then normalized by an existing baseline slope
+image. New hot pixels are identified as those with normalized signal
+rates above a ``hot_threshold`` value. Similarly, pixels with
+normalized signal rates below a ``dead_threshold`` are flagged as new
+dead pixels.
```

Comment:
I would say yes, unless the instrument teams want to do some more complex analysis when looking for bad pixels (e.g. line/function fitting to the ramp files). But in this case we have to make the slope images for the dark monitor anyway, so rather than make them again in a bad pixel monitor, why not work on them now.

Reply:
That's great!